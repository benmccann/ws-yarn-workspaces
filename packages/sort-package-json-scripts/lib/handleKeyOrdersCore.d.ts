import type { ISortPackageJsonScriptsOptions, ISortPackageJsonScriptsOptionsRequired } from './types';
export declare function handleKeyOrdersCore(names: string[], { otherScriptNames, defaultNpmScriptsOrder, omitKeyFn, sortKeyFn, }: ISortPackageJsonScriptsOptions | ISortPackageJsonScriptsOptionsRequired): string[];
export default handleKeyOrdersCore;
