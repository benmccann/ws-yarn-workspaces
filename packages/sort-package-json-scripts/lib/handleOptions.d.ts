import type { ISortPackageJsonScriptsOptions, ISortPackageJsonScriptsOptionsRequired } from './types';
export declare function handleOptions(opts?: ISortPackageJsonScriptsOptions): ISortPackageJsonScriptsOptionsRequired;
export default handleOptions;
