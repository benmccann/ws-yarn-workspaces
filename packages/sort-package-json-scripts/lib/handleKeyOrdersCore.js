"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleKeyOrdersCore = void 0;
function handleKeyOrdersCore(names, { otherScriptNames, defaultNpmScriptsOrder, omitKeyFn, sortKeyFn, }) {
    const prefixable = new Set();
    const keymap = {};
    const addToPrefixable1 = (value, { key, omitted, name, list, }) => {
        var _a;
        keymap[key] = ((_a = keymap[key]) !== null && _a !== void 0 ? _a : []);
        keymap[key].push(name);
        prefixable.add(value);
        list.push(value);
    };
    const keys = names
        .sort(sortKeyFn)
        .reduce((a, name) => {
        const { key, omitted } = omitKeyFn(name);
        if (defaultNpmScriptsOrder.has(name)) {
            addToPrefixable1(name, {
                key,
                omitted,
                name,
                list: a.list2,
            });
        }
        if (defaultNpmScriptsOrder.has(key)) {
            addToPrefixable1(key, {
                key,
                omitted,
                name,
                list: a.list2,
            });
        }
        else if (defaultNpmScriptsOrder.has(omitted)) {
            addToPrefixable1(omitted, {
                key,
                omitted,
                name,
                list: a.list2,
            });
        }
        else if (otherScriptNames.has(name)) {
            addToPrefixable1(name, {
                key,
                omitted,
                name,
                list: a.list1,
            });
        }
        else if (otherScriptNames.has(key)) {
            addToPrefixable1(key, {
                key,
                omitted,
                name,
                list: a.list1,
            });
        }
        else if (otherScriptNames.has(omitted) || key !== omitted) {
            addToPrefixable1(omitted, {
                key,
                omitted,
                name,
                list: a.list1,
            });
        }
        else {
            a.list1.push(name);
        }
        return a;
    }, {
        list1: [],
        list2: [],
    });
    const order = [
        ...defaultNpmScriptsOrder.values(),
        ...keys.list2,
        ...keys.list1,
    ].reduce((order, key) => {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j;
        if (prefixable.has(key)) {
            order.push(...((_c = (_b = (_a = keymap[`pre${key}`]) === null || _a === void 0 ? void 0 : _a.sort) === null || _b === void 0 ? void 0 : _b.call(_a)) !== null && _c !== void 0 ? _c : []));
            order.push(...((_f = (_e = (_d = keymap[key]) === null || _d === void 0 ? void 0 : _d.sort) === null || _e === void 0 ? void 0 : _e.call(_d)) !== null && _f !== void 0 ? _f : []));
            order.push(...((_j = (_h = (_g = keymap[`post${key}`]) === null || _g === void 0 ? void 0 : _g.sort) === null || _h === void 0 ? void 0 : _h.call(_g)) !== null && _j !== void 0 ? _j : []));
        }
        else {
            order.push(key);
        }
        return order;
    }, []);
    return order;
}
exports.handleKeyOrdersCore = handleKeyOrdersCore;
exports.default = handleKeyOrdersCore;
//# sourceMappingURL=handleKeyOrdersCore.js.map