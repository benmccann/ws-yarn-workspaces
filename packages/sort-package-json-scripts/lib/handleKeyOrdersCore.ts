import type { ISortPackageJsonScriptsOptions, ISortPackageJsonScriptsOptionsRequired } from './types';

export function handleKeyOrdersCore(names: string[], {
	otherScriptNames,
	defaultNpmScriptsOrder,
	omitKeyFn,
	sortKeyFn,
}: ISortPackageJsonScriptsOptions | ISortPackageJsonScriptsOptionsRequired)
{
	const prefixable = new Set<string>()
	const keymap = {} as Record<string, string[]>

	const addToPrefixable1 = (value: string, {
		key,
		omitted,
		name,
		list,
	}: {
		key: string,
		omitted: string,
		name: string,
		list: string[],
	}) =>
	{
		keymap[key] = (keymap[key] ?? []);
		keymap[key].push(name);

		prefixable.add(value);
		list.push(value);
	}

	const keys = names
		.sort(sortKeyFn)
		.reduce((a, name) =>
		{
			const { key, omitted } = omitKeyFn(name)

			if (defaultNpmScriptsOrder.has(name))
			{
				addToPrefixable1(name, {
					key,
					omitted,
					name,
					list: a.list2,
				})
			}
			else if (defaultNpmScriptsOrder.has(key))
			{
				addToPrefixable1(key, {
					key,
					omitted,
					name,
					list: a.list2,
				})
			}
			else if (defaultNpmScriptsOrder.has(omitted))
			{
				addToPrefixable1(omitted, {
					key,
					omitted,
					name,
					list: a.list2,
				})
			}
			else if (otherScriptNames.has(name))
			{
				addToPrefixable1(name, {
					key,
					omitted,
					name,
					list: a.list1,
				})
			}
			else if (otherScriptNames.has(key))
			{
				addToPrefixable1(key, {
					key,
					omitted,
					name,
					list: a.list1,
				})
			}
			else if (otherScriptNames.has(omitted) || key !== omitted)
			{
				addToPrefixable1(omitted, {
					key,
					omitted,
					name,
					list: a.list1,
				})
			}
			else if (name !== key)
			{
				addToPrefixable1(key, {
					key,
					omitted,
					name,
					list: a.list1,
				})
			}
			else
			{
				a.list1.push(name)
			}

			return a
		}, {
			list1: [] as string[],
			list2: [] as string[],
		})
	;

	const order = [
		...defaultNpmScriptsOrder.values(),
		...keys.list2,
		...keys.list1,
	].reduce((order, key) =>
		{
			if (prefixable.has(key))
			{
				order.push(`pre${key}`);
				order.push(...(keymap[`pre${key}`]?.sort?.() ?? []));
				order.push(key);
				order.push(...(keymap[key]?.sort?.() ?? []));
				order.push(`post${key}`);
				order.push(...(keymap[`post${key}`]?.sort?.() ?? []));
			}
			else
			{
				order.push(key)
			}

			return order
		}, [] as string[],
	)

	return order
}

export default handleKeyOrdersCore
