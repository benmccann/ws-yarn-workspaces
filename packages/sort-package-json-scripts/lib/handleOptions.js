"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = handleOptions;
const util_1 = require("./util");
function handleOptions(opts) {
    var _a, _b, _c;
    return {
        ...(opts !== null && opts !== void 0 ? opts : {}),
        otherScriptNames: (_a = opts === null || opts === void 0 ? void 0 : opts.otherScriptNames) !== null && _a !== void 0 ? _a : util_1.otherScriptNames,
        defaultNpmScriptsOrder: (_b = opts === null || opts === void 0 ? void 0 : opts.defaultNpmScriptsOrder) !== null && _b !== void 0 ? _b : util_1.defaultNpmScriptsOrder,
        omitKeyFn: (_c = opts === null || opts === void 0 ? void 0 : opts.omitKeyFn) !== null && _c !== void 0 ? _c : util_1.omitKey,
        sortKeyFn: opts === null || opts === void 0 ? void 0 : opts.sortKeyFn,
    };
}
exports.default = handleOptions;
//# sourceMappingURL=handleOptions.js.map