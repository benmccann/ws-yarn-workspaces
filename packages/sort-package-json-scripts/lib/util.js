"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.otherNpmScriptsOrder = exports.defaultNpmScriptsOrder = exports.omitKey = void 0;
function omitKey(name) {
    const key = name
        .replace(/^[_:\-]+/, '')
        .split(/[_:\-]+/)[0]
        .replace(/\d+$/, '');
    const omitted = key
        .replace(/^(?:pre|post)/, '');
    return {
        name,
        key,
        omitted,
    };
}
exports.omitKey = omitKey;
exports.defaultNpmScriptsOrder = new Set([
    'start',
    'dev',
    'restart',
    'stop',
    'coverage',
    'lint',
    'test',
    'install',
    'uninstall',
    'build',
    'link',
    'npm',
    'prepublish',
    'prepare',
    'prepublishOnly',
    'prepack',
    'pack',
    'postpack',
    'publish',
    'postpublish',
    'shrinkwrap',
    'version',
]);
exports.otherNpmScriptsOrder = new Set([
    'prettier',
]);
//# sourceMappingURL=util.js.map