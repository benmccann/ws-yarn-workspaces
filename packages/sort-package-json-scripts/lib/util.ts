export function trimKey(name: string, skipNumber?: boolean)
{
	return name
		.replace(/^[_:\-]+/, '')
		.replace(skipNumber ? /[_:\-]+$/ : /[\d_:\-]+$/, '')
}

export function firstPartKey(name: string)
{
	let key = trimKey(name);
	let first = key.split(/[_:\-]+/)[0];

	if (first === '')
	{
		key = trimKey(name, true);
		first = key.split(/[_:\-]+/)[0];
	}

	return first
}

/**
 * omit key logic
 */
export function omitKey(name: string)
{
	const key = firstPartKey(name);
	const omitted = key
		.replace(/^(?:pre|post)/, '')
	;

	return {
		/**
		 * input name
		 */
		name,
		/**
		 * omit name and only keep first part
		 */
		key,
		/**
		 * omit key with pre / post
		 */
		omitted,
	}
}

/**
 * group / sore scripts order, by default is follow npm lifecycle scripts
 */
export const defaultNpmScriptsOrder = new Set([

	'start',
	'dev',
	'restart',
	'stop',

	'coverage',
	'lint',
	'test',

	'install',
	'uninstall',

	'build',

	'link',

	'npm',

	'prepublish',
	'prepare',
	'prepublishOnly',
	'prepack',
	'pack',
	'postpack',
	'publish',

	'postpublish',

	'shrinkwrap',

	'version',

])

/**
 * avoid omitKey wrong parse script name (e.g. prettier
 */
export const otherScriptNames = new Set([

	'prettier',

])

