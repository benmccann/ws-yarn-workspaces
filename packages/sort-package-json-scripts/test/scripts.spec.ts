/// <reference types="jest" />

import sortPackageJsonScripts from '../index';
import { defaultNpmScriptsOrder } from '../lib/util';

it('sort base', function ()
{
	const fixture = {
		test: 'node test.js',
		multiply: '2 * 3', // between p(ostinstall) and install
		watch: 'watch things',
		prewatch: 'echo "about to watch"',
		postinstall: 'echo "Installed"',
		start: 'node server.js',
		posttest: 'abc',
		pretest: 'xyz',
		postprettier: 'echo "so pretty"',
		preprettier: 'echo "not pretty"',
		prettier: 'prettier -l "**/*.js"',
		prepare: 'npm run build',
		'pre-fetch-info': 'foo',
	}

	const expected = {
		start: 'node server.js',
		pretest: 'xyz',
		test: 'node test.js',
		posttest: 'abc',
		postinstall: 'echo "Installed"',
		prepare: 'npm run build',
		multiply: '2 * 3',
		preprettier: 'echo "not pretty"',
		prettier: 'prettier -l "**/*.js"',
		postprettier: 'echo "so pretty"',
		'pre-fetch-info': 'foo',
		prewatch: 'echo "about to watch"',
		watch: 'watch things'
	}

	let actual = sortPackageJsonScripts(fixture)

	expect(actual).toStrictEqual(expected);

	expectMatchSnapshot(actual);
	expectStringify(actual, expected)
});

it('sort base v2', function ()
{
	const fixture = {
		test: 'node test.js',
		multiply: '2 * 3', // between p(ostinstall) and install
		watch: 'watch things',
		prewatch: 'echo "about to watch"',
		postinstall: 'echo "Installed"',
		start: 'node server.js',
		posttest: 'abc',
		pretest: 'xyz',
		_postprettier: 'echo "so pretty"',
		preprettier_: 'echo "not pretty"',
		prettier_: 'prettier -l "**/*.js"',
		prepare: 'npm run build',
		'pre-fetch-info': 'foo',
	}

	const expected = {
		start: 'node server.js',
		pretest: 'xyz',
		test: 'node test.js',
		posttest: 'abc',
		postinstall: 'echo "Installed"',
		prepare: 'npm run build',
		preprettier_: 'echo "not pretty"',
		prettier_: 'prettier -l "**/*.js"',
		_postprettier: 'echo "so pretty"',
		multiply: '2 * 3',
		'pre-fetch-info': 'foo',
		prewatch: 'echo "about to watch"',
		watch: 'watch things'
	}

	let actual = sortPackageJsonScripts(fixture)

	expect(actual).toStrictEqual(expected);

	expectMatchSnapshot(actual);
	expectStringify(actual, expected)
});

it('sort yarn-tool scripts', function ()
{
	const fixture = {
		"coverage": "npx nyc yarn run test",
		"lint": "npx eslint **/*.ts",
		"sort-package-json": "npx yarn-tool sort",
		"test:mocha": "npx mocha --require ts-node/register \"!(node_modules)/**/*.{test,spec}.{ts,tsx}\"",
		"tsc:default": "tsc -p tsconfig.json",
		"ncu": "npx yarn-tool ncu -u",
		"npm:publish": "npm publish",
		"postpublish_": "git commit -m \"chore(release): publish\" .",
		"prepublish:lockfile": "npx sync-lockfile .",
		"prepublishOnly_": "yarn run ncu && yarn run sort-package-json && yarn run test",
		"test": "echo \"Error: no test specified\" && exit 1",
		"tsc:esm": "tsc -p tsconfig.esm.json"
	}

	const fixture2 = {
		"coverage": "npx nyc yarn run test",
		"lint": "npx eslint **/*.ts",
		"ncu": "npx yarn-tool ncu -u",
		"npm:publish": "npm publish",
		"tsc:default": "tsc -p tsconfig.json",
		"_postpublish_": "git commit -m \"chore(release): publish\" .",
		"prepublish:lockfile": "npx sync-lockfile .",
		"tsc:esm": "tsc -p tsconfig.esm.json",
		"_prepublishOnly:": "yarn run ncu && yarn run sort-package-json && yarn run test",
		"sort-package-json": "npx yarn-tool sort",
		"test-": "echo \"Error: no test specified\" && exit 1",
		"test:mocha": "npx mocha --require ts-node/register \"!(node_modules)/**/*.{test,spec}.{ts,tsx}\"",


	}

	let actual = sortPackageJsonScripts(fixture)
	let actual2 = sortPackageJsonScripts(fixture2)

	expectMatchSnapshot(actual);
	expectMatchSnapshot(actual2);
});

it('should defaultNpmScriptsOrder', function ()
{
	let keys = [...defaultNpmScriptsOrder.values()];

	let source = keys.slice().reverse()
		.reduce((a, k, i) => {
			a[k] = i.toString();
			return a
		}, {} as Record<string, string>);

	let actual = sortPackageJsonScripts(source)

	expect(Object.keys(actual)).toStrictEqual(keys)

	expectMatchSnapshot(actual)

	source = keys.slice()
		.reduce((a, k, i) => {
			a[k] = i.toString();
			return a
		}, {} as Record<string, string>);

	actual = sortPackageJsonScripts(source)

	expect(Object.keys(actual)).toStrictEqual(keys)

	expectMatchSnapshot(actual)
});

function expectStringify(actual, expected)
{
	expect(actual).toStrictEqual(expected);
	expect(JSON.stringify(actual, null, 2)).toStrictEqual(JSON.stringify(expected, null, 2));
}

function expectMatchSnapshot(actual)
{
	expect(actual).toMatchSnapshot();
	expect(Object.keys(actual)).toMatchSnapshot();
	expect(JSON.stringify(actual, null, 2)).toMatchSnapshot();
}
