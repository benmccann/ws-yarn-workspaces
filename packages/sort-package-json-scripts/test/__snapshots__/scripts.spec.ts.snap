// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should defaultNpmScriptsOrder 1`] = `
Object {
  "build": "12",
  "coverage": "17",
  "dev": "20",
  "install": "14",
  "link": "11",
  "lint": "16",
  "npm": "10",
  "pack": "5",
  "postpack": "4",
  "postpublish": "2",
  "prepack": "6",
  "prepare": "8",
  "prepublish": "9",
  "prepublishOnly": "7",
  "publish": "3",
  "restart": "19",
  "shrinkwrap": "1",
  "start": "21",
  "stop": "18",
  "test": "15",
  "uninstall": "13",
  "version": "0",
}
`;

exports[`should defaultNpmScriptsOrder 2`] = `
Array [
  "start",
  "dev",
  "restart",
  "stop",
  "coverage",
  "lint",
  "test",
  "install",
  "uninstall",
  "build",
  "link",
  "npm",
  "prepublish",
  "prepare",
  "prepublishOnly",
  "prepack",
  "pack",
  "postpack",
  "publish",
  "postpublish",
  "shrinkwrap",
  "version",
]
`;

exports[`should defaultNpmScriptsOrder 3`] = `
"{
  \\"start\\": \\"21\\",
  \\"dev\\": \\"20\\",
  \\"restart\\": \\"19\\",
  \\"stop\\": \\"18\\",
  \\"coverage\\": \\"17\\",
  \\"lint\\": \\"16\\",
  \\"test\\": \\"15\\",
  \\"install\\": \\"14\\",
  \\"uninstall\\": \\"13\\",
  \\"build\\": \\"12\\",
  \\"link\\": \\"11\\",
  \\"npm\\": \\"10\\",
  \\"prepublish\\": \\"9\\",
  \\"prepare\\": \\"8\\",
  \\"prepublishOnly\\": \\"7\\",
  \\"prepack\\": \\"6\\",
  \\"pack\\": \\"5\\",
  \\"postpack\\": \\"4\\",
  \\"publish\\": \\"3\\",
  \\"postpublish\\": \\"2\\",
  \\"shrinkwrap\\": \\"1\\",
  \\"version\\": \\"0\\"
}"
`;

exports[`should defaultNpmScriptsOrder 4`] = `
Object {
  "build": "9",
  "coverage": "4",
  "dev": "1",
  "install": "7",
  "link": "10",
  "lint": "5",
  "npm": "11",
  "pack": "16",
  "postpack": "17",
  "postpublish": "19",
  "prepack": "15",
  "prepare": "13",
  "prepublish": "12",
  "prepublishOnly": "14",
  "publish": "18",
  "restart": "2",
  "shrinkwrap": "20",
  "start": "0",
  "stop": "3",
  "test": "6",
  "uninstall": "8",
  "version": "21",
}
`;

exports[`should defaultNpmScriptsOrder 5`] = `
Array [
  "start",
  "dev",
  "restart",
  "stop",
  "coverage",
  "lint",
  "test",
  "install",
  "uninstall",
  "build",
  "link",
  "npm",
  "prepublish",
  "prepare",
  "prepublishOnly",
  "prepack",
  "pack",
  "postpack",
  "publish",
  "postpublish",
  "shrinkwrap",
  "version",
]
`;

exports[`should defaultNpmScriptsOrder 6`] = `
"{
  \\"start\\": \\"0\\",
  \\"dev\\": \\"1\\",
  \\"restart\\": \\"2\\",
  \\"stop\\": \\"3\\",
  \\"coverage\\": \\"4\\",
  \\"lint\\": \\"5\\",
  \\"test\\": \\"6\\",
  \\"install\\": \\"7\\",
  \\"uninstall\\": \\"8\\",
  \\"build\\": \\"9\\",
  \\"link\\": \\"10\\",
  \\"npm\\": \\"11\\",
  \\"prepublish\\": \\"12\\",
  \\"prepare\\": \\"13\\",
  \\"prepublishOnly\\": \\"14\\",
  \\"prepack\\": \\"15\\",
  \\"pack\\": \\"16\\",
  \\"postpack\\": \\"17\\",
  \\"publish\\": \\"18\\",
  \\"postpublish\\": \\"19\\",
  \\"shrinkwrap\\": \\"20\\",
  \\"version\\": \\"21\\"
}"
`;

exports[`sort base 1`] = `
Object {
  "multiply": "2 * 3",
  "postinstall": "echo \\"Installed\\"",
  "postprettier": "echo \\"so pretty\\"",
  "posttest": "abc",
  "pre-fetch-info": "foo",
  "prepare": "npm run build",
  "preprettier": "echo \\"not pretty\\"",
  "pretest": "xyz",
  "prettier": "prettier -l \\"**/*.js\\"",
  "prewatch": "echo \\"about to watch\\"",
  "start": "node server.js",
  "test": "node test.js",
  "watch": "watch things",
}
`;

exports[`sort base 2`] = `
Array [
  "start",
  "pretest",
  "test",
  "posttest",
  "postinstall",
  "prepare",
  "multiply",
  "preprettier",
  "prettier",
  "postprettier",
  "pre-fetch-info",
  "prewatch",
  "watch",
]
`;

exports[`sort base 3`] = `
"{
  \\"start\\": \\"node server.js\\",
  \\"pretest\\": \\"xyz\\",
  \\"test\\": \\"node test.js\\",
  \\"posttest\\": \\"abc\\",
  \\"postinstall\\": \\"echo \\\\\\"Installed\\\\\\"\\",
  \\"prepare\\": \\"npm run build\\",
  \\"multiply\\": \\"2 * 3\\",
  \\"preprettier\\": \\"echo \\\\\\"not pretty\\\\\\"\\",
  \\"prettier\\": \\"prettier -l \\\\\\"**/*.js\\\\\\"\\",
  \\"postprettier\\": \\"echo \\\\\\"so pretty\\\\\\"\\",
  \\"pre-fetch-info\\": \\"foo\\",
  \\"prewatch\\": \\"echo \\\\\\"about to watch\\\\\\"\\",
  \\"watch\\": \\"watch things\\"
}"
`;

exports[`sort base v2 1`] = `
Object {
  "_postprettier": "echo \\"so pretty\\"",
  "multiply": "2 * 3",
  "postinstall": "echo \\"Installed\\"",
  "posttest": "abc",
  "pre-fetch-info": "foo",
  "prepare": "npm run build",
  "preprettier_": "echo \\"not pretty\\"",
  "pretest": "xyz",
  "prettier_": "prettier -l \\"**/*.js\\"",
  "prewatch": "echo \\"about to watch\\"",
  "start": "node server.js",
  "test": "node test.js",
  "watch": "watch things",
}
`;

exports[`sort base v2 2`] = `
Array [
  "start",
  "pretest",
  "test",
  "posttest",
  "postinstall",
  "prepare",
  "preprettier_",
  "prettier_",
  "_postprettier",
  "multiply",
  "pre-fetch-info",
  "prewatch",
  "watch",
]
`;

exports[`sort base v2 3`] = `
"{
  \\"start\\": \\"node server.js\\",
  \\"pretest\\": \\"xyz\\",
  \\"test\\": \\"node test.js\\",
  \\"posttest\\": \\"abc\\",
  \\"postinstall\\": \\"echo \\\\\\"Installed\\\\\\"\\",
  \\"prepare\\": \\"npm run build\\",
  \\"preprettier_\\": \\"echo \\\\\\"not pretty\\\\\\"\\",
  \\"prettier_\\": \\"prettier -l \\\\\\"**/*.js\\\\\\"\\",
  \\"_postprettier\\": \\"echo \\\\\\"so pretty\\\\\\"\\",
  \\"multiply\\": \\"2 * 3\\",
  \\"pre-fetch-info\\": \\"foo\\",
  \\"prewatch\\": \\"echo \\\\\\"about to watch\\\\\\"\\",
  \\"watch\\": \\"watch things\\"
}"
`;

exports[`sort base v2: {
  "postinstall": "echo \\"Installed\\"",
  "multiply": "2 * 3",
  "pre-fetch-info": "foo",
  "prepare": "npm run build",
  "preprettier_": "echo \\"not pretty\\"",
  "prettier_": "prettier -l \\"**/*.js\\"",
  "_postprettier": "echo \\"so pretty\\"",
  "start": "node server.js",
  "pretest": "xyz",
  "test": "node test.js",
  "posttest": "abc",
  "prewatch": "echo \\"about to watch\\"",
  "watch": "watch things"
} 1`] = `
"{
  \\"start\\": \\"node server.js\\",
  \\"pretest\\": \\"xyz\\",
  \\"test\\": \\"node test.js\\",
  \\"posttest\\": \\"abc\\",
  \\"postinstall\\": \\"echo \\\\\\"Installed\\\\\\"\\",
  \\"prepare\\": \\"npm run build\\",
  \\"preprettier_\\": \\"echo \\\\\\"not pretty\\\\\\"\\",
  \\"prettier_\\": \\"prettier -l \\\\\\"**/*.js\\\\\\"\\",
  \\"_postprettier\\": \\"echo \\\\\\"so pretty\\\\\\"\\",
  \\"multiply\\": \\"2 * 3\\",
  \\"pre-fetch-info\\": \\"foo\\",
  \\"prewatch\\": \\"echo \\\\\\"about to watch\\\\\\"\\",
  \\"watch\\": \\"watch things\\"
}"
`;

exports[`sort base: {
  "postinstall": "echo \\"Installed\\"",
  "multiply": "2 * 3",
  "pre-fetch-info": "foo",
  "prepare": "npm run build",
  "preprettier": "echo \\"not pretty\\"",
  "prettier": "prettier -l \\"**/*.js\\"",
  "postprettier": "echo \\"so pretty\\"",
  "start": "node server.js",
  "pretest": "xyz",
  "test": "node test.js",
  "posttest": "abc",
  "prewatch": "echo \\"about to watch\\"",
  "watch": "watch things"
} 1`] = `
"{
  \\"start\\": \\"node server.js\\",
  \\"pretest\\": \\"xyz\\",
  \\"test\\": \\"node test.js\\",
  \\"posttest\\": \\"abc\\",
  \\"postinstall\\": \\"echo \\\\\\"Installed\\\\\\"\\",
  \\"prepare\\": \\"npm run build\\",
  \\"multiply\\": \\"2 * 3\\",
  \\"preprettier\\": \\"echo \\\\\\"not pretty\\\\\\"\\",
  \\"prettier\\": \\"prettier -l \\\\\\"**/*.js\\\\\\"\\",
  \\"postprettier\\": \\"echo \\\\\\"so pretty\\\\\\"\\",
  \\"pre-fetch-info\\": \\"foo\\",
  \\"prewatch\\": \\"echo \\\\\\"about to watch\\\\\\"\\",
  \\"watch\\": \\"watch things\\"
}"
`;

exports[`sort base: {
  "start": "node server.js",
  "pretest": "xyz",
  "test": "node test.js",
  "posttest": "abc",
  "postinstall": "echo \\"Installed\\"",
  "prepare": "npm run build",
  "multiply": "2 * 3",
  "preprettier": "echo \\"not pretty\\"",
  "prettier": "prettier -l \\"**/*.js\\"",
  "postprettier": "echo \\"so pretty\\"",
  "pre-fetch-info": "foo",
  "prewatch": "echo \\"about to watch\\"",
  "watch": "watch things"
} 1`] = `
"{
  \\"start\\": \\"node server.js\\",
  \\"pretest\\": \\"xyz\\",
  \\"test\\": \\"node test.js\\",
  \\"posttest\\": \\"abc\\",
  \\"postinstall\\": \\"echo \\\\\\"Installed\\\\\\"\\",
  \\"prepare\\": \\"npm run build\\",
  \\"multiply\\": \\"2 * 3\\",
  \\"preprettier\\": \\"echo \\\\\\"not pretty\\\\\\"\\",
  \\"prettier\\": \\"prettier -l \\\\\\"**/*.js\\\\\\"\\",
  \\"postprettier\\": \\"echo \\\\\\"so pretty\\\\\\"\\",
  \\"pre-fetch-info\\": \\"foo\\",
  \\"prewatch\\": \\"echo \\\\\\"about to watch\\\\\\"\\",
  \\"watch\\": \\"watch things\\"
}"
`;

exports[`sort base: {
  "test": "node test.js",
  "multiply": "2 * 3",
  "watch": "watch things",
  "prewatch": "echo \\"about to watch\\"",
  "postinstall": "echo \\"Installed\\"",
  "start": "node server.js",
  "posttest": "abc",
  "pretest": "xyz",
  "postprettier": "echo \\"so pretty\\"",
  "preprettier": "echo \\"not pretty\\"",
  "prettier": "prettier -l \\"**/*.js\\"",
  "prepare": "npm run build",
  "pre-fetch-info": "foo"
} 1`] = `
"{
  \\"start\\": \\"node server.js\\",
  \\"pretest\\": \\"xyz\\",
  \\"test\\": \\"node test.js\\",
  \\"posttest\\": \\"abc\\",
  \\"postinstall\\": \\"echo \\\\\\"Installed\\\\\\"\\",
  \\"prepare\\": \\"npm run build\\",
  \\"multiply\\": \\"2 * 3\\",
  \\"preprettier\\": \\"echo \\\\\\"not pretty\\\\\\"\\",
  \\"prettier\\": \\"prettier -l \\\\\\"**/*.js\\\\\\"\\",
  \\"postprettier\\": \\"echo \\\\\\"so pretty\\\\\\"\\",
  \\"pre-fetch-info\\": \\"foo\\",
  \\"prewatch\\": \\"echo \\\\\\"about to watch\\\\\\"\\",
  \\"watch\\": \\"watch things\\"
}"
`;

exports[`sort yarn-tool scripts 1`] = `
Object {
  "coverage": "npx nyc yarn run test",
  "lint": "npx eslint **/*.ts",
  "ncu": "npx yarn-tool ncu -u",
  "npm:publish": "npm publish",
  "postpublish_": "git commit -m \\"chore(release): publish\\" .",
  "prepublish:lockfile": "npx sync-lockfile .",
  "prepublishOnly_": "yarn run ncu && yarn run sort-package-json && yarn run test",
  "sort-package-json": "npx yarn-tool sort",
  "test": "echo \\"Error: no test specified\\" && exit 1",
  "test:mocha": "npx mocha --require ts-node/register \\"!(node_modules)/**/*.{test,spec}.{ts,tsx}\\"",
  "tsc:default": "tsc -p tsconfig.json",
  "tsc:esm": "tsc -p tsconfig.esm.json",
}
`;

exports[`sort yarn-tool scripts 2`] = `
Array [
  "coverage",
  "lint",
  "test",
  "test:mocha",
  "npm:publish",
  "prepublish:lockfile",
  "prepublishOnly_",
  "postpublish_",
  "ncu",
  "sort-package-json",
  "tsc:default",
  "tsc:esm",
]
`;

exports[`sort yarn-tool scripts 3`] = `
"{
  \\"coverage\\": \\"npx nyc yarn run test\\",
  \\"lint\\": \\"npx eslint **/*.ts\\",
  \\"test\\": \\"echo \\\\\\"Error: no test specified\\\\\\" && exit 1\\",
  \\"test:mocha\\": \\"npx mocha --require ts-node/register \\\\\\"!(node_modules)/**/*.{test,spec}.{ts,tsx}\\\\\\"\\",
  \\"npm:publish\\": \\"npm publish\\",
  \\"prepublish:lockfile\\": \\"npx sync-lockfile .\\",
  \\"prepublishOnly_\\": \\"yarn run ncu && yarn run sort-package-json && yarn run test\\",
  \\"postpublish_\\": \\"git commit -m \\\\\\"chore(release): publish\\\\\\" .\\",
  \\"ncu\\": \\"npx yarn-tool ncu -u\\",
  \\"sort-package-json\\": \\"npx yarn-tool sort\\",
  \\"tsc:default\\": \\"tsc -p tsconfig.json\\",
  \\"tsc:esm\\": \\"tsc -p tsconfig.esm.json\\"
}"
`;

exports[`sort yarn-tool scripts 4`] = `
Object {
  "_postpublish_": "git commit -m \\"chore(release): publish\\" .",
  "_prepublishOnly:": "yarn run ncu && yarn run sort-package-json && yarn run test",
  "coverage": "npx nyc yarn run test",
  "lint": "npx eslint **/*.ts",
  "ncu": "npx yarn-tool ncu -u",
  "npm:publish": "npm publish",
  "prepublish:lockfile": "npx sync-lockfile .",
  "sort-package-json": "npx yarn-tool sort",
  "test-": "echo \\"Error: no test specified\\" && exit 1",
  "test:mocha": "npx mocha --require ts-node/register \\"!(node_modules)/**/*.{test,spec}.{ts,tsx}\\"",
  "tsc:default": "tsc -p tsconfig.json",
  "tsc:esm": "tsc -p tsconfig.esm.json",
}
`;

exports[`sort yarn-tool scripts 5`] = `
Array [
  "coverage",
  "lint",
  "test-",
  "test:mocha",
  "npm:publish",
  "prepublish:lockfile",
  "_prepublishOnly:",
  "_postpublish_",
  "ncu",
  "sort-package-json",
  "tsc:default",
  "tsc:esm",
]
`;

exports[`sort yarn-tool scripts 6`] = `
"{
  \\"coverage\\": \\"npx nyc yarn run test\\",
  \\"lint\\": \\"npx eslint **/*.ts\\",
  \\"test-\\": \\"echo \\\\\\"Error: no test specified\\\\\\" && exit 1\\",
  \\"test:mocha\\": \\"npx mocha --require ts-node/register \\\\\\"!(node_modules)/**/*.{test,spec}.{ts,tsx}\\\\\\"\\",
  \\"npm:publish\\": \\"npm publish\\",
  \\"prepublish:lockfile\\": \\"npx sync-lockfile .\\",
  \\"_prepublishOnly:\\": \\"yarn run ncu && yarn run sort-package-json && yarn run test\\",
  \\"_postpublish_\\": \\"git commit -m \\\\\\"chore(release): publish\\\\\\" .\\",
  \\"ncu\\": \\"npx yarn-tool ncu -u\\",
  \\"sort-package-json\\": \\"npx yarn-tool sort\\",
  \\"tsc:default\\": \\"tsc -p tsconfig.json\\",
  \\"tsc:esm\\": \\"tsc -p tsconfig.esm.json\\"
}"
`;
